lib/KBox/src/KMessage.h

```diff
@@ -191,6 +202,92 @@ class IMUMessage: public KMessage {
     };
 };
 
+class NAVMessage: public KMessage { //RIGM added for autopilot functionality
+private:
+    String label; //not sure we need a label
+    bool apMode, apHeadingMode, apWaypointMode, apDodgeMode;
+    double currentHeading, targetHeading, courseToWaypoint, rudderSensorPosition, targetRudderPosition, rudderCommandSent;
+    
+public:
+    NAVMessage(String label, bool apMode, bool apHeadingMode, bool apWaypointMode, bool apDodgeMode, double currentHeading, double targetHeading, double courseToWaypoint, double rudderSensorPosition) :
+    label(label), apMode(apMode), apHeadingMode(apHeadingMode), apWaypointMode(apWaypointMode), apDodgeMode(apDodgeMode),
+    currentHeading(currentHeading),targetHeading(targetHeading),courseToWaypoint(courseToWaypoint),
+    rudderSensorPosition(rudderSensorPosition){};
+    
+    void accept(KVisitor &v) const {
+        v.visit(*this);
+    };
+    
+    String getLabel() const {
+        return label;
+    };
+    
+    bool getApMode() const {
+        return apMode;
+    };
+    
+    bool getApHeadingMode() const {
+        return apHeadingMode;
+    };
+    
+    bool getApWaypointMode() const {
+        return apWaypointMode;
+    };
+    
+    bool getApDodgeMode() const {
+        return apDodgeMode;
+    };
+    
+    /*
+     * Courses in Degrees, therefore using doubles:
```

Comment:
I tried changing the NAVMessage to store values as radians and adding the DegToRad and RadToDeg conversions on messages, but the PID code will no longer compile:
PID headingPID(RadToDeg(&apCurrentHeading), &apTargetRudderPosition, (RadToDeg)&apTargetHeading, (double)P_Param, (double)I_Param, (double)D_Param, (int)REVERSE);  error: cannot convert 'double*' to 'double' for argument '1' to 'double RadToDeg(double)'
I'm probably not doing this right.  For now I will leave it the way it was, which works.